### PR TITLE
Fix after not showing when not truncated

### PR DESCRIPTION
**Pull request by gaygenius**
This addresses problems with the fixes from [this branch](https://github.com/swergas/jQuery.dotdotdot/tree/fix-paragraph-and-read-more-not-showing):
- Preserves markup on if ellipsis is added to the last element in some cases where the container had caused the height to be exceeded.
- Avoids errors with `!$e.is( o.after ) && !$e.find( o.after ).length`

### CODE REVIEW

File: src/js/jquery.dotdotdot.js [hunk 1]
```diff
@@ -169,6 +169,10 @@
 						.append( orgContent )
 						.attr( 'style', $dot.data( 'dotdotdot-style' ) || '' )
 						.data( 'dotdotdot', false );
+					if ( typeof fn == 'function' )
+					{
+						fn.call( $dot[ 0 ] );
```
Comment (saponifi3d) [inline src/js/jquery.dotdotdot.js]: we don't want to pass any additional context to this function? any arguments to pass through?

File: src/js/jquery.dotdotdot.js [hunk 2]
```diff
@@ -387,7 +395,9 @@
 
 							if ( !isTruncated )
 							{
-								$e.detach();
```
Comment (saponifi3d) [inline src/js/jquery.dotdotdot.js]: do we not need to detach anymore because we aren't attaching it if we can't?

**Comment by saponifi3d** [comment]
Generally LGTM just a couple clarification questions to make sure i understand :) :rocket:

File: src/js/jquery.dotdotdot.js (hunk 1 above)
Comment (gaygenius) [inline src/js/jquery.dotdotdot.js]: No more arguments were needed in my use case.  As you can see, there are specific events to retrieve the original content and whether it was truncated.

In this case, the client has requested that the dotdotdot behavior be removed.  What do you think should be passed to such a client?

Comment (saponifi3d) [inline src/js/jquery.dotdotdot.js]: Not sure - just wanted to make sure :) - the other thing i'd want to make sure is that `$dot[0]` is a jQuery element, not an DOM object. Looks like it *might* be the latter?

File: src/js/jquery.dotdotdot.js (hunk 2 above)
Comment (gaygenius) [inline src/js/jquery.dotdotdot.js]: This bug fix came from the repo I forked from.  Please see [this related PR](https://github.com/BeSite/jQuery.dotdotdot/pull/46) and the explanation of the bug.

Comment (saponifi3d) [inline src/js/jquery.dotdotdot.js]: :+1: kewl - thanks!

File: src/js/jquery.dotdotdot.js (hunk 1 above)
Comment (gaygenius) [inline src/js/jquery.dotdotdot.js]: I think you’re right that it’s a DOM element.  I picked that because every other callback in this library passes that as the `this` parameter.  It turns out that in my use case I bind `this` and therefore I don’t use the `this` passed here, but I still wanted to be consistent with the other callbacks so as to avoid confusion.

Comment (saponifi3d) [inline src/js/jquery.dotdotdot.js]: okay - if it's consistent with theirs :+1:

File: src/js/jquery.dotdotdot.js [hunk 3]
```diff
@@ -387,8 +395,17 @@
...
 							{
-								$e.detach();
-								isTruncated = true;
+								// force truncation of last text node within $e
+								var lastTextNode = findLastTextNode( $e, $d );
+								if ( lastTextNode ) ellipsisElement( $( lastTextNode ), $d, $i, o, after );
+
+								// if $e contains that last text node, then we want after link immediately after it
+								if ( $.contains( $e[0], lastTextNode ) ) {
+									addAfter( $e );
```
Comment (letitia) [inline src/js/jquery.dotdotdot.js]: Can you clarify the difference between `$e` and `$elem`?  I'm just trying to envision if there'd be a case where we wouldn't want the `after` to come after `$e`.

Comment (gaygenius) [inline src/js/jquery.dotdotdot.js]: This broke when `$e` referred to an `<hr>` element, which cannot contain text.  In that case, the addAfter (which does a jQuery `.append`) did not result in a link being shown to the user.

**Comment by saponifi3d** [comment]
Still LGTM :dancers: - was helpful to dive into it a bit more :+1: :rocket:

File: src/js/jquery.dotdotdot.js (hunk 3 above)
Comment (letitia) [inline src/js/jquery.dotdotdot.js]: Ok, thanks.

**Comment by gaygenius** [comment]
It’s been way too long since I last looked at this so I’m closing it without merging. Some investigation could determine whether this fix is still needed.

**Comment by saponifi3d** [comment]
👋
